### ISSUE TITLE
New Component Specific User Mods

### ISSUE THREAD

**Issue by mvertens** [ESMCI/cime#1520]
We need the ability for a component to set user mods for other components in regular expression matching capability. As an example CAM needs to make sure that all orbital settings are set to specific values for ALL aquaplanet compsets. The current usermods attachment to a compset would mean that effectively 60 new compsets need to be added to ensure this dependence with the current scheme. This PR would eliminate the need to introduce many compsets via the introduction of the following simple new entry in CAM's config_component.xml.

 ```
<entry id="CAM_USER_MODS">
    <type>char</type>
    <valid_values></valid_values>
    <values>
      <value compset="DOCN%SAQP">$SRCROOT/components/cam/cime_config/user_mods/foo</value>
      <value compset="DOCN%PAQP">$SRCROOT/components/cam/cime_config/user_mods/foo</value>
    </values>
  </entry>
```

**Comment by billsacks** [comment]
@jedwards4b and @mvertens - should this have been closed with #1523 ?

**Comment by mvertens** [comment]
Yes - I think this should have been closed. @jedwards4b - could you please
verify this.
